src/test/java/com/group2/catanbackend/gamelogic/GameLogicControllerExpandedTest.java

```diff
@@ -322,6 +324,77 @@ void testFalseAccusationGetsProperlyPunished(){
         assertEquals(3, totalResources);
     }
 
+    @Test
+    void testMakeTradeOfferMoveWithBankWorkingRight(){
+        //p1 init with [1,1,1,2,1]
+        player1.adjustResources(new int[]{3,3,3,2,3});
+        int[] offeredResources = {4,0,0,0,0};
+        int[] wantedResources = {0,0,0,0,1};
+        ArrayList<Integer> sendToPlayer = new ArrayList<>();
+        MakeTradeOfferMoveDto makeTradeOfferMoveDto = new MakeTradeOfferMoveDto(negate(offeredResources), wantedResources, sendToPlayer);
+        gameLogicController.setSetupPhase(false);
+        gameLogicController.makeMove(makeTradeOfferMoveDto, player1);
+        assertArrayEquals(new int[]{0,4,4,4,5}, player1.getResources());
+    }
+
+    @Test
+    void testMakeTradingWithPlayer(){
+        //p1 init with [1,1,1,2,1], p2: [1,1,2,2,0]
+        int[] offeredResources = {1, 1, 0, 0, 0};
+        int[] wantedResources = {0, 0, 2, 2, 0};
+        ArrayList<Integer> sendToPlayer = new ArrayList<>();
+        sendToPlayer.add(player2.getInGameID());
+        gameLogicController.setSetupPhase(false);
+        gameLogicController.makeMove(new MakeTradeOfferMoveDto(negate(offeredResources), wantedResources, sendToPlayer), player1);
+        TradeOfferDto tradeOfferDto = new TradeOfferDto(negate(wantedResources), offeredResources, player1.toInGamePlayerDto());
+        tradeOfferDto.setEventType(null);
+        gameLogicController.makeMove(new AcceptTradeOfferMoveDto(tradeOfferDto), player2);
+        assertArrayEquals(new int[]{0,0,3,4,1}, player1.getResources());
+        assertArrayEquals(new int[]{2,2,0,0,0}, player2.getResources());
+    }
+
+    @Test
+    void testTradingOverDiceRoll(){
```

Comment:
again unclear name, should be something like "tradePersistsOverDiceBeingRolled" or something